psp-bugos/RandomDataGenerator/RandomDataGenerator.cs

```diff
@@ -2,41 +2,48 @@ namespace psp_bugos.RandomDataGenerator;
 
 public class RandomDataGenerator : IRandomDataGenerator
 {
-    public T GenerateValues<T>() where T : class, new()
+    public T GenerateValues<T>(Guid? id = null) where T : class, new()
     {
         var result = new T();
         var properties = typeof(T).GetProperties();
         var random = new Random();
         foreach (var property in properties)
         {
-            switch (property.PropertyType)
+            if (property.Name != "Id")
             {
-                case { } t when t == typeof(string):
-                    property.SetValue(result, $"Test {property.Name}");
-                    break;
-                case { } t when t == typeof(int):
-                    property.SetValue(result, random.Next());
-                    break;
-                case { } t when t == typeof(decimal):
-                    property.SetValue(result, new decimal(random.NextDouble()));
-                    break;
-                case { } t when t == typeof(DateTime):
-                    property.SetValue(result, DateTime.Now);
-                    break;
-                case { } t when t == typeof(bool):
-                    property.SetValue(result, random.Next(0, 2) == 1);
-                    break;
-                case { } t when t == typeof(Guid):
-                    property.SetValue(result, Guid.NewGuid());
-                    break;
-                case { } t when t == typeof(Enum):
-                    property.SetValue(result,
-                        Enum.GetValues(property.PropertyType)
-                            .GetValue(random.Next(0, Enum.GetValues(property.PropertyType).Length)));
-                    break;
-                default:
-                    property.SetValue(result, null);
-                    break;
+                switch (property.PropertyType)
+                {
+                    case { } t when t == typeof(string):
+                        property.SetValue(result, $"Test {property.Name}");
+                        break;
```

Comment:
NIT: This switch statement is too much verbose.

This one takes only 2 lines for each case, doesn't introduce new (unused) variables:
```suggestion
                var type = property.PropertyType;
                
                if (type == typeof(string))
                    property.SetValue(result, $"Test {property.Name}");
```

Another way how this could be simplified (but doesn't solve long scroll issue):
```suggestion
                var type = property.PropertyType;
                switch (type)
                {
                    case { } when type == typeof(string):
                        property.SetValue(result, $"Test {property.Name}");
                        break;
```

Reply:
used first option to shorten `GenerateValues<T>` method.